src/store/undoRedo/getChangedKeys.ts

```diff
@@ -0,0 +1,18 @@
+/* eslint-disable import/prefer-default-export, @typescript-eslint/no-explicit-any */
+export function getChangedKeys(a: any = {}, b: any = {}): string[] {
+  const changedKeys: string[] = []
+
+  Object.keys({...a as object, ...b as object})
+    .forEach((key) => {
+      const aVal = (a as Record<string, any>)[key]
+      const bVal = (b as Record<string, any>)[key]
+
+      if (typeof aVal === 'object' && typeof bVal === 'object') {
+        changedKeys.push(...getChangedKeys(aVal as object, bVal as object).map((k) => `${key}.${k}`))
+      } else if (aVal !== bVal) {
+        changedKeys.push(key)
+      }
+    })
+
+  return changedKeys
+}
```

Comment:
Should we use stricter types for the args so that `getChangedKeys` is not called with primitives?
I've played with this a bit and also noticed that `=== object` passes through `null`.
Suggestion:
```
// eslint-disable-next-line import/prefer-default-export
export function getChangedKeys(a: object, b: object): string[] {
  const changedKeys: string[] = []

  Object.keys({...a, ...b})
    .forEach((key) => {
      const aVal = (a as Record<string, unknown>)[key]
      const bVal = (b as Record<string, unknown>)[key]

      if (typeof aVal === 'object' && typeof bVal === 'object' && aVal !== null && bVal !== null) {
        changedKeys.push(...getChangedKeys(aVal, bVal).map((k) => `${key}.${k}`))
      } else if (aVal !== bVal) {
        changedKeys.push(key)
      }
    })

  return changedKeys
}
```
Luckily, the null doesn't currently break things as spreading it in `Object.keys({...null, ...null})` gives an empty array.
So the suggested code doesn't change much (just marginally better typing?)